Mathlib/Analysis/NormedSpace/Alternating.lean

```diff
@@ -0,0 +1,556 @@
+/-
+Copyright (c) 2020 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel, Yury Kudryashov, Heather Macbeth, Patrick Massot
+-/
+import Mathlib.Topology.Algebra.Module.Alternating.Basic
+import Mathlib.Analysis.NormedSpace.Multilinear.Basic
+
+/-!
+# Operator norm on the space of continuous alternating maps
+
+-/
+
+noncomputable section
+
+open scoped BigOperators NNReal
+open Finset Metric
+
+/-!
+### Type variables
+
+We use the following type variables in this file:
+
+* `𝕜` : a `NontriviallyNormedField`;
+* `ι`, `ι'` : finite index types with decidable equality;
+* `E`, `E₁` : families of normed vector spaces over `𝕜` indexed by `i : ι`;
+* `E'` : a family of normed vector spaces over `𝕜` indexed by `i' : ι'`;
+* `Ei` : a family of normed vector spaces over `𝕜` indexed by `i : Fin (Nat.succ n)`;
+* `G`, `G'` : normed vector spaces over `𝕜`.
+-/
+
+universe u v v' wE wE₁ wE' wEi wG wG'
+variable {𝕜 : Type u} {n : ℕ}
+  {E : Type wE} {E₁ : Type wE₁} {E' : Type wE'} {Ei : Type wEi}
+  {G : Type wG} {G' : Type wG'} {ι : Type v} {ι' : Type v'}
+  [Fintype ι] [Fintype ι'] [NontriviallyNormedField 𝕜]
+  [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  [NormedAddCommGroup E₁] [NormedSpace 𝕜 E₁]
+  [NormedAddCommGroup E'] [NormedSpace 𝕜 E']
+  [NormedAddCommGroup Ei] [NormedSpace 𝕜 Ei]
+  [NormedAddCommGroup G] [NormedSpace 𝕜 G]
+  [NormedAddCommGroup G'] [NormedSpace 𝕜 G']
+
+/-!
+### Continuity properties of alternating maps
+
+We relate continuity of alternating maps to the inequality `‖f m‖ ≤ C * ∏ i, ‖m i‖`, in
+both directions. Along the way, we prove useful bounds on the difference `‖f m₁ - f m₂‖`.
+-/
+namespace AlternatingMap
+
+variable (f : E [Λ^ι]→ₗ[𝕜] G)
+
+/-- If an alternating map in finitely many variables on normed spaces satisfies the inequality
+`‖f m‖ ≤ C * ∏ i, ‖m i‖` on a shell `ε / ‖c‖ < ‖m i‖ < ε` for some positive number `ε`
+and an elements `c : 𝕜`, `1 < ‖c‖`, then it satisfies this inequality for all `m`. -/
+lemma bound_of_shell {ε : ℝ} {C : ℝ} (hε : 0 < ε) {c : 𝕜} (hc : 1 < ‖c‖)
+    (hf : ∀ m : ι → E, (∀ i, ε / ‖c‖ ≤ ‖m i‖) → (∀ i, ‖m i‖ < ε) → ‖f m‖ ≤ C * ∏ i, ‖m i‖)
+    (m : ι → E) : ‖f m‖ ≤ C * ∏ i, ‖m i‖ :=
+  f.toMultilinearMap.bound_of_shell (fun _ ↦ hε) (fun _ ↦ hc) hf m
```

Comment:
Do you think it would make sense to change this (and other results) to `MultilinearMapClass.bound_of_shell`, so that we don't have to state it twice?

"yes but not in this PR" is a fair answer too.

Reply:
Yes but not in this PR.